addons/mrp/models/mrp_production.py

```diff
@@ -1405,12 +1408,20 @@ def _get_quantity_produced_issues(self):
     def _action_generate_backorder_wizard(self, quantity_issues):
         ctx = self.env.context.copy()
         lines = []
+        ask_create_backorder = always_create_backorder = self.env['mrp.production']
         for order in quantity_issues:
-            lines.append((0, 0, {
-                'mrp_production_id': order.id,
-                'to_backorder': True
-            }))
-        ctx.update({'default_mrp_production_ids': self.ids, 'default_mrp_production_backorder_line_ids': lines})
+            if order.picking_type_id.create_backorder == 'ask' or (order.picking_type_id.create_backorder == 'never' and order.product_id.tracking == 'serial'):
```

Comment:
It should be handle in _pre_button_mark_done. Keep _action_generate_backorder_wizard only handle the creation of the wizard

Reply:
Better in _get_quantity_produced_issues to be able to call in WO